docs/using-flatpak.rst

```diff
@@ -0,0 +1,103 @@
+Using Flatpak
+=============
+
+This page provides an introduction to the most common commands needed to use Flatpak. It is not intended to be exhaustive or to cover all the options for each command (a full list of all the commands and their options can be found in `the command reference <command-reference.html>`_).
+
+.. note::
+  Flatpak commands can be run either per-user or system-wide. All the examples in this guide use the default system-wide behavior.
+
+Remotes
+-------
+
+Remotes are the repositories from which applications and runtimes can be installed.
+
+List remotes
+````````````
+
+To list the remotes that you have configured on your system, run::
+
+  $ flatpak remotes
+
+This gives a list of the existing remotes that have been added. The list indicates whether each remote has been added per-user or system-wide.
+
+Add a remote
+````````````
+
+Adding a remote allows you to search and list its contents, and to install applications from it. The most convenient way to add a remote is by using a ``.flatpakrepo`` file, which includes both the details of the remote and its GPG key::
+
+ $ flatpak remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo
+
+Here, ``flathub`` is the local name that is given to the remote. The URL points to the remote's ``.flatpakrepo`` file. ``--if-not-exists`` prevents an existing remote from being overwritten.
```

Comment:
It doesn't prevent overwritting, it will actually just error out otherwise.

Reply:
`--if-not-exists` just prevents the command from erroring out?

Reply:
Yes.

Reply:
Sounds like that should be default behaviour, perhaps? Every example I see of that command uses --if-not-exists. Anyway, I'll update the branch for now...

Reply:
I think this flag is mostly useful for scripts. A script may either want to know if a repo didn't get added or doesn't care. For users I think either behavior is fine and it saying it already existed is harmless.

Reply:
> For users I think either behavior is fine and it saying it already existed is harmless.

It's an extra argument though, which is more work for users.

Reply:
I've fixed the explanation in the PR branch for now.